asciidoctorj-api/src/main/java/org/asciidoctor/extension/BaseProcessor.java

```diff
@@ -0,0 +1,216 @@
+package org.asciidoctor.extension;
+
+import org.asciidoctor.Options;
+import org.asciidoctor.ast.*;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+public class BaseProcessor implements Processor {
+
+    private Processor delegate;
+
+    public BaseProcessor() {
+        this(new HashMap<>());
+    }
+
+    public BaseProcessor(Processor delegate) {
+        this(delegate, new HashMap<>());
+    }
+
+    public BaseProcessor(Map<String, Object> config) {
+        // FIXME: Use a strategy to resolve the processor
+        this(ServiceLoader.load(Processor.class).iterator().next(), config);
```

Comment:
I think we should introduce a "service" to resolve the implementation so the logic can be easily tested.
It will also ensure that the logic is the same everywhere.

Reply:
@robertpanzer We still need to use a proper resolution mechanism here 🏷 
As long as someone does not provide another implementation we should be fine 😓